src/contracts/engine/transactions/sellAndBurnMln.test.ts

```diff
@@ -0,0 +1,217 @@
+import { BigNumber } from 'bignumber.js';
+import { appendDecimals } from '@melonproject/token-math/token';
+import { createQuantity } from '@melonproject/token-math/quantity';
+import { getPrice } from '@melonproject/token-math/price';
+import { initTestEnvironment, getGlobalEnvironment } from '~/utils/environment';
+import { Address } from '~/utils/types';
+import { deploy as deployEngine, sellAndBurnMln } from '..';
+import {
+  deploy as deployToken,
+  approve,
+  getToken,
+  balanceOf,
+} from '~/contracts/dependencies/token';
+import { deploy as deployFeed, update } from '~/contracts/prices';
+import {
+  increaseTime,
+  deployAndGetContract,
+  getContract,
+} from '~/utils/solidity';
+import { Contracts } from '~/Contracts';
+
+const shared: any = {};
+
+// TODO: use the actual sellAndBurnMln typescript function rather than testing like this
+beforeAll(async () => {
+  await initTestEnvironment();
+  shared.env = getGlobalEnvironment();
+  shared.accounts = await shared.env.eth.getAccounts();
+  const wethAddress = await deployToken('ETH');
+  shared.mln = await deployAndGetContract('dependencies/token/BurnableToken', [
```

Comment:
Hmm, I understand that it might be more time-consuming to write all calls and transactions first and it's just faster to directly get the web3 contract instance. Which is fine in the sense of rapid prototyping.

But we should aim for having all relevant calls and transactions as stand alone files. The `callFactory` and `transactionFactory` can be as easy as that:

```ts
import { callFactory } from '~/utils/solidity';
import { Contracts } from '~/Contracts';

const getHub = callFactory('getHub', Contracts.Spoke);

export { getHub };

// ---

import { transactionFactory } from '~/utils/solidity';
import { Contracts } from '~/Contracts';

export const continueCreation = transactionFactory(
  'continueCreation',
  Contracts.FundFactory,
);
```

The advantage of this model is that we can slowly refactor and enhance these functions then as we see need, like adding guards and post-processing, without changing the consumers at all or only a little.

What I want to say: Consider using the factories to quickly create simple method files which are easily extensible. But using the contracts directly in the tests is fine for now I think. But only in the tests. `getContract` & `deployAndGetContract` should'nt be exported at all.

Reply:
Ok yea that sounds good to me :)